src/components/NewMessage/NewMessage.vue

```diff
@@ -699,9 +706,13 @@ export default {
 				// Also remove the message to be replied for this conversation
 				this.chatExtrasStore.removeParentIdToReply(this.token)
 
-				this.broadcast
-					? await this.broadcastMessage(this.token, temporaryMessage.message)
-					: await this.postMessage(this.token, temporaryMessage, options)
+				// Delegate sending to HOC, unless it's upload dialog
+				// If upload reaches there, then it is a normal message without any shares
+				if (this.dialog && !this.upload) {
+					await this.submitMessage(this.token, temporaryMessage, options)
+				} else {
+					await this.postMessage(this.token, temporaryMessage, options)
+				}
```

Comment:
It should still delegate sending the message to the parent when it's upload (also dialog) 
[The case where you have text in upload editor but removed all files]